### PR TITLE
Performance Profiler: add badge to highest impact insights

### PR DESCRIPTION
**Pull request by gcsecsey**
Closes https://github.com/Automattic/dotcom-forge/issues/9019

## Proposed Changes

* add a badge to the highest impact insights

## Why are these changes being made?

https://github.com/Automattic/dotcom-forge/issues/9019 

## Testing Instructions

* Go to `/speed-test` and start a new test, or go to a results page eg. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzA1N30.7KXaBL1qcm0y0GqvM8ymJ0pbST9ls6P09_hri8EaKSI`
* Check the styling of the insights table on both mobile and desktop.

| Viewport | Before | After |
|--------|--------|--------|
| Mobile | CleanShot 2024-09-05 at 17 42 15@2x | CleanShot 2024-09-05 at 17 40 12@2x |
| Desktop | CleanShot 2024-09-05 at 17 40 46@2x | CleanShot 2024-09-05 at 17 41 13@2x | 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/performance-profiler/components/metrics-insight/insight-header.tsx [hunk 1]
```diff
@@ -38,6 +38,9 @@ export const InsightHeader: React.FC< InsightHeaderProps > = ( props ) => {
 					<span className={ clsx( 'value', { [ type ]: true } ) }> { value }</span>
 				</span>
 			) }
+			{ type === 'fail' && (
+				<span className={ clsx( 'impact fail', { 'is-mobile': isMobile() } ) }>High Impact</span>
+			) }
```
Comment (WBerredo) [inline client/performance-profiler/components/metrics-insight/insight-header.tsx]: Is this how we define if we should have a `high impact` badge? I understood it was a different check,

**Review by WBerredo — DISMISSED** [review]
We also need some changes on the font. The following should do the trick

```css
font-size: $font-body-small;
color: #000;
```

File: client/performance-profiler/components/metrics-insight/insight-header.tsx (hunk 1 above)
Comment (gavande1) [reply client/performance-profiler/components/metrics-insight/insight-header.tsx]: We should show an audit as `High Impact` when the `metricSavings` key is not empty.

Comment (gcsecsey) [reply client/performance-profiler/components/metrics-insight/insight-header.tsx]: Oh sorry, I missed that this should be the trigger for it. I've added it in ce4f4785c7

File: client/data/site-profiler/types.ts [hunk 2]
```diff
@@ -187,6 +187,7 @@ export interface PerformanceMetricsItemQueryResponse {
 	type: 'warning' | 'fail';
 	displayValue?: string;
 	details?: PerformanceMetricsDetailsQueryResponse;
+	metricSavings: { FCP: number; LCP: number };
```
Comment (gavande1) [inline client/data/site-profiler/types.ts]: The metricSavings can be FCP, LCP, CLS, or INP; all are optional. Don't you want to add them? 

CleanShot 2024-09-06 at 17 19 58@2x

**Review by gavande1 — APPROVED** [review]
LGTM 👍

**Pull request merged**